src/plugins/uk.ac.kcl.mdeoptimise.ui/src/uk/ac/kcl/ui/classpath/MDEOClassPathContainer.xtend

```diff
@@ -21,15 +21,14 @@ class MDEOClassPathContainer implements IClasspathContainer {
 		 }
 	}
 	
-	override getDescription() '''
-		Core MDEOptimiser libraries
-	'''
+	override getDescription() '''Core MDEOptimiser Libraries'''
+		
 	
 	override getKind() {
 		IClasspathContainer.K_SYSTEM
 	}
 	
 	override getPath() { 
-		new Path("MDEOptimiser/basic")
+		MDEOClassPathContainerInitializer.MDEO_LIBRARY_PATH

```

Comment:
This doesn't seem right: The spec requires these paths to be two part: the first part is the name of the container, the second part gives additional hints (so that different versions of the container could be provided). I had defined a `basic` version, which would be installed. The container requested would be labelled as `default`.

Reply:
I have looked at the source code of other projects and this seems to be the way they do it. The two parts are the namespace followed by the container name suffix.

For example:
https://github.com/eclipse/xtext-xtend/blob/b375d6b677bb72bcb7a1eb8012b55a9c913d6ab8/org.eclipse.xtend.ide/src/org/eclipse/xtend/ide/buildpath/XtendContainerInitializer.java#L28

I thought a similar approach would be good.

Reply:
Just because someone else does it, too, doesn't mean it's correct. If you look up the documentation on `JavaCore` and `IClassPathContainer`, you will find that they have a very clear idea of what to use. Note that these strings may be exposed to users in the IDE, so should be chosen well.

Reply:
I think this is correct. They suggest using a container ID segment followed by extra segments to be used as hints for calling the container initializer.

The ID should be the namespace of our bundle and the extra segment should ideally be only one that points to the container initializer id, that is going to be used by the ClasspathContainerInitializer extension in plugin.xml.

This is why I used the container identifier constant here to match exactly the container id.

Actually, this container should only be added to the classpath when the validator detects java constraints or objectives in the DSL.

Reply:
But if you look at the examples, they look more like what I did. 

To be honest, as long as the ugly long string doesn't show to the user, I don't care. I have seen at least one place where it can show (namely when you add the container directly in the launch config class path), but there might be others.

Reply:
I think both approaches have to be combined. Looking at the container for the JRE, it seems to be a mix between IDs and specific identifiers for each:

org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8

So perhaps the best way is to have something like this:

uk.ac.kcl.mdeoptimise.MDEO_CONTAINER/dsl - for the DSL functionality, which for now will include everything;

But this will allow us to extend this in the future by adding further identifiers for specific functionality.

So uk.ac.kcl.mdeoptimise.MDEO_CONTAINER becomes the ID and the /dsl is the set of classpath entries required by the DSL to run with java based fitness functions.

Reply:
OK, let's go with that for the moment.